### PR TITLE
Merge from upstream through 2020-07-17

### PR DESCRIPTION
**Pull request by amgleitman**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Merge facebook/react-native's main branch into react-native-macos's main branch through 2020-07-17 23:59:59 PDT (specifically, f00795dc90d05180014eeea4b3215b0166a90692).

These changes were made by repeatedly running git merge <commit-id> and resolving merge conflicts as needed.

## Notes

* e96f1e1d830d20f88f020afa9a34a8f3b710b30e introduces a regression in RNTester that is resolved by e0e33fce531e8faef1922b47321706792fbd9ed9, so this merge includes cherry-picking the latter as well as implementing a similar fix for macOS.

## Validation

* yarn lint
* * ✖ 66 problems (0 errors, 66 warnings)
* yarn flow-check-ios
* yarn flow-check-macos
* yarn flow-check-android
* yarn test
* * Test Suites: 128 passed, 128 total
* * Tests: 1 skipped, 1257 passed, 1258 total
* * Snapshots: 582 passed, 582 total
* RNTester
* * Compiles on macOS and iOS
* * All unit tests within Xcode pass except for WebSocketTest (this is expected for now)
* * No new regressions have been found

### CODE REVIEW

File: Libraries/Text/TextInput/RCTBaseTextInputView.m [hunk 1]
```diff
@@ -381,7 +381,7 @@ - (void)textInputDidBeginEditing
 
   [_eventDispatcher sendTextEventWithType:RCTTextEventTypeFocus
                                  reactTag:self.reactTag
-                                     text:[self.backedTextInputView.attributedText.string copy] // [TODO(macOS Candidate ISS#2710739)
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/RCTBaseTextInputView.m:384]: What's the reason for removing this TODO? Was our fixed pushed upstream?

File: Libraries/Text/TextInput/RCTBaseTextInputView.m [hunk 2]
```diff
@@ -413,17 +413,11 @@ - (BOOL)textInputShouldReturn
   // `onSubmitEditing` is called when "Submit" button
   // (the blue key on onscreen keyboard) did pressed
   // (no connection to any specific "submitting" process).
-#if TARGET_OS_OSX // [TODO(macOS Candidate ISS#2710739)
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/RCTBaseTextInputView.m:416]: Why don't we need this anymore?

File: Libraries/Text/TextInput/RCTBaseTextInputView.m [hunk 3]
```diff
@@ -481,7 +475,7 @@ - (NSString *)textInputShouldChangeText:(NSString *)text inRange:(NSRange)range
     }
   }
 
-  NSString *previousText = [backedTextInputView.attributedText.string copy] ?: @""; // TODO(OSS Candidate ISS#2710739)
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/RCTBaseTextInputView.m:484]: Was this macOS change pushed upstream and that's why we're removing our TODO?

File: Libraries/Text/TextInput/RCTBaseTextInputView.m [hunk 4]
```diff
@@ -528,7 +522,7 @@ - (void)textInputDidChange
 
   if (_onChange) {
     _onChange(@{
-       @"text": [self.attributedText.string copy], // [TODO(macOS Candidate ISS#2710739)
```
Comment (HeyImChris) [inline Libraries/Text/TextInput/RCTBaseTextInputView.m:531]: Was this macOS change pushed upstream and that's why we're removing our TODO?

File: Libraries/Text/TextInput/RCTBaseTextInputView.m (hunk 1 above)
Comment (HeyImChris) [reply Libraries/Text/TextInput/RCTBaseTextInputView.m:384]: Same question for the other removed TODOs in this file

Comment (amgleitman) [reply Libraries/Text/TextInput/RCTBaseTextInputView.m:384]: Yes, with 15dda0ab5a491dcc83539f9ef32c9896be41074a.

File: Libraries/Text/TextInput/RCTBaseTextInputView.m (hunk 3 above)
Comment (amgleitman) [reply Libraries/Text/TextInput/RCTBaseTextInputView.m:484]: Yes, with 15dda0ab5a491dcc83539f9ef32c9896be41074a.

File: Libraries/Text/TextInput/RCTBaseTextInputView.m (hunk 4 above)
Comment (amgleitman) [reply Libraries/Text/TextInput/RCTBaseTextInputView.m:531]: Yes, with 15dda0ab5a491dcc83539f9ef32c9896be41074a.

File: Libraries/Text/TextInput/RCTBaseTextInputView.m (hunk 2 above)
Comment (amgleitman) [reply Libraries/Text/TextInput/RCTBaseTextInputView.m:416]: This slipped through. I believe we still need this.

**Review by HeyImChris — APPROVED** [review]

**Pull request merged**